codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2PushPromiseFrameTest.java

```diff
@@ -0,0 +1,219 @@
+                Http2FrameStream newPushFrameStream = newStream();
+                Http2PushPromiseFrame pushPromiseFrame = new DefaultHttp2PushPromiseFrame(pushRequestHeaders);
+                pushPromiseFrame.stream(receivedFrame.stream());
+                pushPromiseFrame.pushStream(newPushFrameStream);
+                ctx.writeAndFlush(pushPromiseFrame).sync();
+
+                contentMap.put(newPushFrameStream.id(), "Meow, I am Pushed via HTTP/2");
+
+                // Write headers for actual request
+                Http2Headers http2Headers = new DefaultHttp2Headers();
+                http2Headers.status("200");
+                http2Headers.add("push", "false");
+                Http2HeadersFrame headersFrame = new DefaultHttp2HeadersFrame(http2Headers, false);
+                headersFrame.stream(receivedFrame.stream());
+                ctx.writeAndFlush(headersFrame).sync();
+
+                // Write Data of actual request
+                Http2DataFrame dataFrame = new DefaultHttp2DataFrame(Unpooled.wrappedBuffer("Meow".getBytes()), true);
+                dataFrame.stream(receivedFrame.stream());
+                ctx.writeAndFlush(dataFrame).sync();
+            } else if (msg instanceof Http2PriorityFrame) {
+                Http2PriorityFrame priorityFrame = (Http2PriorityFrame) msg;
+                String content = contentMap.get(priorityFrame.stream().id());
+                if (content == null) {
+                    ctx.writeAndFlush(new DefaultHttp2GoAwayFrame(Http2Error.REFUSED_STREAM));
+                    return;
+                }
+
+                // Write headers for Priority request
+                Http2Headers http2Headers = new DefaultHttp2Headers();
+                http2Headers.status("200");
+                http2Headers.add("push", "true");
+                Http2HeadersFrame headersFrame = new DefaultHttp2HeadersFrame(http2Headers, false);
+                headersFrame.stream(priorityFrame.stream());
+                ctx.writeAndFlush(headersFrame).sync();
+
+                // Write Data of Priority request
+                Http2DataFrame dataFrame = new DefaultHttp2DataFrame(Unpooled.wrappedBuffer(content.getBytes()), true);
+                dataFrame.stream(priorityFrame.stream());
+                ctx.writeAndFlush(dataFrame).sync();
+            }
+        }
+    }
+
+    private static final class ClientHandler extends Http2ChannelDuplexHandler {
+
+        @Override
+        public void channelActive(ChannelHandlerContext ctx) throws InterruptedException {
+            Http2Headers http2Headers = new DefaultHttp2Headers();
+            http2Headers.path("/")
+                    .authority("localhost")
+                    .method("GET")
+                    .scheme("https");
+
+            Http2HeadersFrame headersFrame = new DefaultHttp2HeadersFrame(http2Headers, true);
+            headersFrame.stream(newStream());
+            ctx.writeAndFlush(headersFrame).sync();
```

Comment:
never call `sync()` in the eventloop. Remove this call and also remove the `throws...`

Reply:
Then I get NPE if I don't block it with something. That's why I had `Thread#sleep`.

Reply:
show me the stack trace

Reply:
Tried adding a listener. PTAL.